1st chunk:
../at-spi2-core-2.48.0/atk-adaptor/bridge.c:138:8: warning: implicit declaration of function 'strncasecmp' is invalid in C99 [-Wimplicit-function-declaration]
  if (!strncasecmp (property, "action.", 7))
../at-spi2-core-2.48.0/atk-adaptor/bridge.c:183:12: warning: implicit declaration of function 'strcasecmp' is invalid in C99 [-Wimplicit-function-declaration]
      if (!strcasecmp (dp->name, member))

2nd chunk:
XXX push upstream; this is what gdk-pixbuf is doing

Index: meson.build
--- meson.build.orig
+++ meson.build
@@ -8,7 +8,7 @@ project('at-spi2-core', 'c',
         ],
         meson_version: '>= 0.63.0')
 
-add_project_arguments([ '-D_POSIX_C_SOURCE=200809L', '-D_DEFAULT_SOURCE' ], language: 'c')
+add_project_arguments([ '-D_POSIX_C_SOURCE=200809L', '-D_BSD_SOURCE' ], language: 'c')
 
 fs = import('fs')
 
@@ -87,16 +87,10 @@ if host_machine.system() == 'darwin'
   common_ldflags += [ '-compatibility_version', darwin_versions[0], '-current_version', darwin_versions[1]]
 endif
 
-# Functions
-checked_funcs = [
-  'bind_textdomain_codeset',
-]
-
-foreach f: checked_funcs
-  if cc.has_function(f)
-    at_spi_conf.set('HAVE_' + f.underscorify().to_upper(), 1)
-  endif
-endforeach
+intl_dep = cc.find_library('intl', required: false)                             
+if cc.has_function('bind_textdomain_codeset', prefix: '#include <libintl.h>', dependencies: intl_dep)
+  at_spi_conf.set('HAVE_BIND_TEXTDOMAIN_CODESET', 1)                        
+endif 
 
 root_inc = include_directories('.')
 atk_inc = include_directories('atk')
